### PR TITLE
dev, v3.0: update TiDB v4.0 roadmap and future plan

### PR DESCRIPTION
**Pull request by yikeke**
### What is changed, added or deleted? 

update TiDB v4.0 roadmap and future plan

### What is the related PR or file link(s)? 
https://github.com/pingcap/docs-cn/pull/1676/
https://github.com/pingcap/docs-cn/pull/1816

### Which version does your change affect? 

dev, v3.0

### CODE REVIEW

File: dev/roadmap.md [hunk 1]
```diff
@@ -1,106 +1,193 @@
...
+
+## TiDB
+
+### TiDB Server
+
+#### Features
+
+* Support TiFlash Storage Engine
+* Support Optimizer Trace
+* Support Multi-Column Statistics
```
Comment (lilin90) [inline dev/roadmap.md]: Please use sentence-style capitalization for items in this doc except for those proper nouns.

Comment (yikeke) [reply dev/roadmap.md]: Updated. PTAL again.

File: dev/roadmap.md [hunk 2]
```diff
@@ -1,106 +1,193 @@
...
+
+### TiDB Server
+
+#### Features
+
+* Common table expression
+* Invisible index
+* Support modifying column types
+* Support second-level partitions for partitioned tables
+* Support interchanging partitioned tables and regular tables
```
Comment (lilin90) [inline dev/roadmap.md]:
```suggestion
* Support conversion between partitioned tables and regular tables
```

File: dev/roadmap.md [hunk 3]
```diff
@@ -1,106 +1,193 @@
...
+* Column-based storage
+* Replicate data from TiKV through Raft learner
+* Snapshot read
+
+### TiSpark
+
+#### Features
+
+* Support batch write
+* Support accessing TiFlash
```
[the remark this answers is not in the document]
Comment (scsldb) [reply dev/roadmap.md]: LGTM

**Comment by scsldb** [comment]
LGTM

**Pull request merged**